library/core/src/str/lossy.rs

```diff
@@ -130,71 +200,15 @@ impl<'a> Iterator for Utf8LossyChunksIter<'a> {
 
         // SAFETY: `valid_up_to <= i` because it is only ever assigned via
         // `valid_up_to = i` and `i` only increases.
-        let (valid, broken) = unsafe { inspected.split_at_unchecked(valid_up_to) };
+        let (valid, invalid) = unsafe { inspected.split_at_unchecked(valid_up_to) };
 
-        Some(Utf8LossyChunk {
+        Some(Utf8Chunk {
             // SAFETY: All bytes up to `valid_up_to` are valid UTF-8.
             valid: unsafe { from_utf8_unchecked(valid) },
-            broken,
+            invalid,
         })
     }
 }
 
-impl fmt::Display for Utf8Lossy {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        // If we're the empty string then our iterator won't actually yield
-        // anything, so perform the formatting manually
-        if self.bytes.is_empty() {
-            return "".fmt(f);
-        }
-
-        for Utf8LossyChunk { valid, broken } in self.chunks() {
-            // If we successfully decoded the whole chunk as a valid string then
-            // we can return a direct formatting of the string which will also
-            // respect various formatting flags if possible.
-            if valid.len() == self.bytes.len() {
-                assert!(broken.is_empty());
-                return valid.fmt(f);
-            }
-
-            f.write_str(valid)?;
-            if !broken.is_empty() {
-                f.write_char(char::REPLACEMENT_CHARACTER)?;
-            }
-        }
-        Ok(())
-    }
-}
-
-impl fmt::Debug for Utf8Lossy {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_char('"')?;
-
-        for Utf8LossyChunk { valid, broken } in self.chunks() {
-            // Valid part.
-            // Here we partially parse UTF-8 again which is suboptimal.
-            {
-                let mut from = 0;
-                for (i, c) in valid.char_indices() {
-                    let esc = c.escape_debug();
-                    // If char needs escaping, flush backlog so far and write, else skip
-                    if esc.len() != 1 {
-                        f.write_str(&valid[from..i])?;
-                        for c in esc {
-                            f.write_char(c)?;
-                        }
-                        from = i + c.len_utf8();
-                    }
-                }
-                f.write_str(&valid[from..])?;
-            }
-
-            // Broken parts of string as hex escape.
-            for &b in broken {
-                write!(f, "\\x{:02x}", b)?;
-            }
-        }
-
-        f.write_char('"')
-    }
-}
```

Comment:
It looks like the new Debug impl is just going to format to the byte slice, rather than the (nicer) view we previously had. Maybe we can keep that nicer Debug impl?

Reply:
The reason I avoided this was that I wasn't sure if the formatting should be `Utf8Chunks("...")` or `"..."`. However, formatting as `"..."` could be helpful to avoid the primary use case of [`bstr::BStr`](https://docs.rs/bstr/latest/bstr/struct.BStr.html).

Reply:
For a Debug impl we typically wrap in the struct name, and I think that makes sense in this case too. We could consider a Display impl that formats as "...", though.

Reply:
The difficulty with using a `Display` impl instead of `Debug` for this is that I don't think there's anywhere else in libstd where `Display` uses byte escapes (except obvious cases such as `EscapeDefault`), so it might be inconsistent. I would prefer to add something similar to `Path::display` that would return a new struct.

Based on your later comment, do you want me to squash commits and keep the PR as-is or add the private method described above and use `debug_struct` to include the struct name?

Reply:
Ah, I didn't quite realize this wasn't addressed in my quick re-review skim. Let's add the private method and use debug_struct; I think that'll be slightly better. That can be squashed with the other commits though.